### ISSUE TITLE
Can not start a new payment intents process from the CP after a customer has started (or completed) one on the same order.

### ISSUE THREAD

**Issue by lukeholder** [craftcms/commerce-stripe#133]
When a store admin tried to start a new payment intents payment on an order that already had a customer start a payment intents process the latter request fails.

Getting the error: "You cannot provide a new payment method to a PaymentIntent when it has a status of requires_capture. You should either capture the remaining funds, or cancel this PaymentIntent and try again with a new PaymentIntent."

The issue is caused by the payment intent being stored by order ID, and not the transaction hash, so additional payments fail thinking they are resuming the original payment intent flow.

**Comment by samhibberd** [comment]
Hi @lukeholder,

I think we are running into an issue related to this, a customer relies heavily on taking additional payments after a order is completed and as a result has a payment status of Partial.

The gateway code is `payment_intent_unexpected_state` and the message is:

> This PaymentIntent's payment_method could not be updated because it has a status of succeeded. You may only update the payment_method of a PaymentIntent with one of the following statuses: requires_payment_method, requires_confirmation, requires_action.

They have only recently migrated over to SCA and started running into this issue.

Moving back to the legacy Charge gateway is not really an option, any timeline for a fix? Or is there a temporary patch?

Many thanks,

Sam

**Comment by samhibberd** [comment]
Hi @lukeholder, is there a timeline for this to be looked at?

**Comment by nickcobley** [comment]
@lukeholder Is this likely to be fixed in the near future? It's going to start causing us some real grief very soon.

**Comment by samhibberd** [comment]
Same here, would be great to see a fix or a timeline for one?

**Comment by joshangell** [comment]
I hit this as we needed to take outstanding balance payments on existing orders - the client edits the order and adds line items or whatever which results in the order showing as partially paid.

I then created a static payment form which of course didn’t work because the payment intent that was stored for the customer could no longer be used.

To that end, I made a quick and dirty controller action that just removed any stored payment intent for an order and I hit that first before hitting `commerce/payments/pay`:

```php
    public function actionPreStaticPayment()
    {
        $this->requirePostRequest();
        $this->requireAcceptsJson();

        $commerce = Commerce::getInstance();

        $orderNumber = Craft::$app->getRequest()->getRequiredParam('orderNumber');

        $order = $commerce->getOrders()->getOrderByNumber($orderNumber);
        if (!$order) {
            return $this->asErrorJson('Could not find order.');
        }

        Craft::$app->getDb()->createCommand()
            ->delete('{{%stripe_paymentintents}}', ['orderId' => $order->id])
            ->execute();

        return $this->asJson([
            'success' => true
        ]);
    }
```

It’s not pretty, but its a useful workaround for now.

**Comment by lukeholder** [comment]
This has been fixed in 2.3.2.2 (We also now have 2.4 out https://github.com/craftcms/commerce-stripe/blob/develop/CHANGELOG.md#240---2021-10-13)
